src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -1,15 +1,1401 @@
+    }
+    this->actor.shape.rot.y = this->actor.posRot.rot.y;
+    yawDiff = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+    if ((ABS(yawDiff) <= temp_t0) && (this->actor.xzDistFromLink < 100.0f)) {
+        if (ABS(this->actor.yDistFromLink) < 150.0f) {
+            if ((globalCtx->gameplayFrames & 1)) {
+                func_80A74E2C(this);
+            } else {
+                func_80A751C8(this);
+            }
+        }
+    }
+    if (func_80A74674(globalCtx, &this->actor) != NULL) {
+        func_80A751C8(this);
+        this->unk_2FC = 1;
+    } else {
+        temp_t0 = this->actor.yawTowardsLink - this->actor.shape.rot.y;
+        if (ABS(temp_t0) > 0x4000) {
+            this->unk_300--;
+            if (this->unk_300 == 0) {
+                func_80A754A0(this);
+            }
+        } else {
+            this->unk_300 = 0x28;
+        }
+    }
+    func_80A745E4(this, globalCtx);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if ((sp30 == (s16)this->skelAnime.animCurrentFrame) || (sp2E == (s16)this->skelAnime.animCurrentFrame)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_IRONNACK_WALK);
+    }
+}
+
+void func_80A74E2C(EnIk* this) {
+    f32 frames = SkelAnime_GetFrameCount(&D_06001C28.genericHeader);
+
+    this->unk_2FF = 1;
+    this->unk_2F8 = 6;
+    this->actor.speedXZ = 0.0f;
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06001C28, 1.5f, 0.0f, frames, 2, -4.0f);
+    EnIk_SetupAction(this, func_80A74EBC);
+}
+
+void func_80A74EBC(EnIk* this, GlobalContext* globalCtx) {
+    Vec3f sp2C;
+
+    if (this->skelAnime.animCurrentFrame == 15.0f) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_IRONNACK_SWING_AXE);
+    } else if (this->skelAnime.animCurrentFrame == 21.0f) {
+        sp2C.x = this->actor.posRot.pos.x + Math_Sins(this->actor.shape.rot.y + 0x6A4) * 70.0f;
+        sp2C.z = this->actor.posRot.pos.z + Math_Coss(this->actor.shape.rot.y + 0x6A4) * 70.0f;
+        sp2C.y = this->actor.posRot.pos.y;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_IRONNACK_HIT_GND);
+        func_8005AA1C(&globalCtx->mainCamera, 2, 0x19, 5);
+        func_800AA000(this->actor.xzDistFromLink, 0xFF, 0x14, 0x96);
+        func_80062CD4(globalCtx, &sp2C);
+    }
+
+    if ((17.0f < this->skelAnime.animCurrentFrame) && (this->skelAnime.animCurrentFrame < 23.0f)) {
+        this->unk_2FE = 1;
+    } else {
+        if ((this->unk_2FB != 0) && (this->skelAnime.animCurrentFrame < 10.0f)) {
+            Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->actor.yawTowardsLink, 1, 0x5DC, 0);
+            this->actor.shape.rot.y = this->actor.posRot.rot.y;
+        }
+        this->unk_2FE = 0;
+    }
+
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime)) {
+        func_80A7506C(this);
+    }
+}
+
+void func_80A7506C(EnIk* this) {
+    f32 frames = SkelAnime_GetFrameCount(&D_060029FC.genericHeader);
+
+    this->unk_2FE = 0;
+    this->unk_2F9 = (s8)frames;
```

Comment:
does s8 temp avoid a cast?

Reply:
Nope, unfortunately not.

Reply:
`unk_2F9` is actually a `u8` but unfortunately storing as / casting to `u8` does a lot of extra floating point operations.